### ISSUE TITLE
VCPKG should have a variable to indicate if the HOST supports x64 native executables

### ISSUE THREAD

**Issue by walbourn** [microsoft/vcpkg#24966]
For a port that has to use a custom Windows EXE *at build or install time*, the only variable that's easy to check is `VCPKG_HOST_IS_WINDOWS`. In almost all cases (except Windows Server) this would indicate that a 32-bit Win32 EXE should run on the "HOST" including Windows on ARM64 which can emulate x86.

What is missing is a `VCPKG_HOST_IS_WINDOWS_X64` which would indicate that the "HOST" is either a Windows x64 native machine -or- that it's a Windows 11 on ARM64 machine which can emulate x64. This is needed to validate that a 64-bit Win32 EXE will run on the "HOST".

**Comment by Neumann-A** [comment]
I don't think this is necessary. Instead it should be controlled by the `host-triplet` you are using.

**Comment by walbourn** [comment]
The host triplet can't indicate if it's Windows 10 on ARM64 that has x86 emulation, but *not* x64 emulation vs. Windows 11 on ARM64 that has both x86 and x64 emulation.

**Comment by Neumann-A** [comment]
> The host triplet can't indicate

You select the host triplet. If you don't give the required information that is on you. Simply use `--host-triplet=arm64-windows` and don't care about emulation at all?

**Comment by walbourn** [comment]
I think you are mis-understanding the scenario...

The issue I'm concerned with is if I'm *authoring* a port that leverages a binary tool that is only provided as x64 native. It's not *built* by the port, and it's used to for *all* target triplets.

For example, in the [cppwinrt](https://github.com/microsoft/vcpkg/pull/24967) port I'm authoring, at the moment the ``cppwinrt.exe`` is an x86 binary that runs on all Windows platforms. However, it's quite reasonable that in future the provider of ``cppwinrt.exe`` might decide to only ship a x64 native version of their 'compiler'. In that case, I do NOT have a way to express that requirement in the port.

**Comment by Neumann-A** [comment]
> For example, in the https://github.com/microsoft/vcpkg/pull/24967 port I'm authoring, at the moment the cppwinrt.exe is an x86 binary that runs on all Windows platforms.

That means you have a `tools` feature with a `supports` expression which says `x86 & x64` and have a host dependency on the port itself with that feature. 

> but it can also work on the arm64-windows host triplet

That is lying to vcpkg. `arm64` means your binary is `arm64`
